jupyterhub/singleuser.py

```diff
@@ -22,8 +23,17 @@
 
 try:
     import notebook
+    use_serverapp = False
+    server_package = 'notebook'
+    app_name = 'notebook.notebookapp'
```

Comment:
Maybe use fully qualified name for the App class `notebook.notebookapp.NotebookApp` rather than having the module name and class name logic in two different places. Then we can import with:

```python
from traitlets.utils.importstring import import_item
app_name = 'notebook.notebookapp.NotebookApp'
...
ServerApp = import_item(app_name)
```